core/server/services/settings/yaml-parser.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const yaml = require('js-yaml'),
+    fs = require('fs-extra'),
+    debug = require('ghost-ignition').debug('settings:yaml-parser'),
+    common = require('../../lib/common');
+
+/**
+ * Takes a YAML file, parses it and returns a JSON Object
+ * @param {String} filePath the absolute path to the file
+ * @param {String} fileName the name of the file incl. extension
+ * @returns {Promise<Object>} parsed
+ * @description Takes a YAML file, parses it and returns a JSON Object.
+ */
+module.exports = function parseYaml(filePath, fileName) {
+    return fs.readFile(filePath, 'utf8')
+        .then((file) => {
+            try {
+                const parsed = yaml.safeLoad(file);
+
+                debug('YAML settings file parsed:', file);
+                return Promise.resolve(parsed);
+            } catch (err) {
+                // Parsing failed, `js-yaml` tells us exactly what and where in the
+                // `reason` property as well as in the message.
+                return Promise.reject(new common.errors.GhostError({
+                    message: common.i18n.t('errors.services.route.settings.error', {file: fileName, context: err.reason}),
+                    context: err.message,
+                    err: err,
+                    help: common.i18n.t('errors.services.route.settings.help', {file: fileName})
+                }));
+            }
+        }).catch({code: 'ENOENT'}, (error) => {
+            return Promise.reject(new common.errors.GhostError({
+                message: 'YAML file in /content/settings directory does not exist',
```

Comment:
The ymlparser shouldn't know that the files are from `content/settings`, but also not high prio.

Reply:
True... true....

Reply:
I made this everywhere variable and return the contentpath as set in the config